### ISSUE TITLE
[Telemetry] Additional Collections telemetry

### ISSUE THREAD

**Issue by ecsmyth** [mozilla-mobile/fenix#9029]
Owner: Product Manager

#### Description & Product Manager / Data Scientist User Story
- As the performance product manager, I want to know how many collections users have and the numbers of tabs saved to the collections, so that we can test raw and perceived performance under representative conditions.
- As a product manager, ..., so that we can improve the UX based on how users use the feature.

#### Hypothesis
- The current UI may result in poor performance and a frustrating experience with a large number of collections or tabs saved per collection.

#### What questions will you answer with this data?
- What variations of our responsiveness tests do we need to ensure we're testing realistic user flows?

#### Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
- User research and exit surveys regularly point to startup time as a point of friction leading to churn. Improving the realism of our test automation improves our ability to detect user-meaningful regressions and make the improvements that matter most to our users.

#### What probes (suggested, if applicable)
- Count of collections
- Histogram of tabs per collection

### Acceptance Criteria (Added by PM)
- Event pings can be queried via re:dash
- We are sending telemetry events for the actions listed in the requirements
- We have documented the telemetry
- We have asked a data steward to [review](https://github.com/mozilla/data-review/blob/master/request.md) the telemetry

**Comment by sraturi** [comment]
Blocked by: https://bugzilla.mozilla.org/show_bug.cgi?id=1648258

**Comment by mcomella** [comment]
Upstream issue was closed as wontfix: sending back to triage.
